File: 242. 有效的字母异位词/Solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        alphas = [0] * 26
        for i in s:
            alphas[ord(i) - ord('a')] += 1
        for i in t:
            alphas[ord(i) - ord('a')] -= 1
        for a in alphas:
            if a != 0:
                return False
        return True



# class Solution:
#     def isAnagram(self, s: str, t: str) -> bool:
#         if len(s) != len(t):
#             return False
#         a_count = Counter(s)
#         t_count = Counter(t)
#         return a_count == t_count

File: 242. 有效的字母异位词/Solution.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] alphas = new int[26];
        for(int i = 0; i < s.length(); i++){
            alphas[s.charAt(i) - 'a'] += 1;
        }
        for(int i = 0; i < t.length(); i++){
            alphas[t.charAt(i) - 'a'] -= 1;
        }
        for(int i = 0; i < alphas.length; i++){
            if(alphas[i] != 0){
                return false;
            }
        }
        return true;
    }
}